system_tests/bloom_test.go

```diff
@@ -17,25 +17,26 @@ import (
 	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/offchainlabs/nitro/arbnode"
 	"github.com/offchainlabs/nitro/solgen/go/mocksgen"
 )
 
 func TestBloom(t *testing.T) {
 	t.Parallel()
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	nodeconfig := arbnode.ConfigDefaultL2Test()
-	nodeconfig.RPC.BloomBitsBlocks = 256
-	nodeconfig.RPC.BloomConfirms = 1
-	l2info, node, client := CreateTestL2WithConfig(t, ctx, nil, nodeconfig, false)
-	defer node.StopAndWait()
+	builder := NewNodeBuilder(ctx).DefaultConfig(false, nil, nil)
+	builder.nodeConfig.RPC.BloomBitsBlocks = 256
+	builder.nodeConfig.RPC.BloomConfirms = 1
+	builder.takeOwnership = false
+	builder.Build(t)
 
-	l2info.GenerateAccount("User2")
+	defer builder.L2.Node.StopAndWait()
```

Comment:
cleanup should be handled better.. I see two options:
1) TestClient has a "cleanup" function (created when building) that does what's necessary
2) build and build2ndNode return cleanup function

1 seems nicer, but 2 makes it easy to check that anyone that calls build will also call cleanup

Reply:
Having cleanup inside TestClient seems like a better option but we can also return the initialized cleanup functions for build and build2ndNode. That way both benefits are met

Reply:
I've implemented a fusion of both ideas to support cases when user wants to change the cleanup function after calling the Build functions